web-service/tests.py

```diff
@@ -16,153 +16,197 @@
-            },
-            {
-                'url': 'http://benfry.com/distellamap/',
-                'rssi': -95,
-                'txpower': -63,
-            },
-            {
-                'url': 'http://en.wikipedia.org/wiki/Le_D%C3%A9jeuner_sur_l%E2%80%99herbe',
-                'rssi': -61,
-                'txpower': -22,
-            },
-            {
-                'url': 'http://sfmoma.org',
-                'rssi': -74,
-                'txpower': -22,
-            },
-        ]
-    }
-
-    result = resolveScanForValues(endpoint, values)
-    print json.dumps(result, indent=4)
-
-################################################################################
-
-def testUrlWhichRedirects(endpoint):
-    values = {
-        'objects': [
-            {
-                'url': 'http://goo.gl/KYvLwO',
-            },
-        ]
-    }
-
-    result = resolveScanForValues(endpoint, values)
-    print json.dumps(result, indent=4)
-
-################################################################################
-
-def testUrlShortener(endpoint):
-    values = {
-        'longUrl': 'http://www.github.com/Google/physical-web'
-    }
-    req = urllib2.Request(endpoint + '/shorten-url', json.dumps(values))
-    response = urllib2.urlopen(req)
-    ret = json.loads(response.read())
-    print ret
-
-################################################################################
-
-def testRefreshUrl(endpoint):
-    params = { 'url': 'https://github.com/google/physical-web' }
-    req = urllib2.Request(endpoint + '/refresh-url?' + urllib.urlencode(params), '')
-    response = urllib2.urlopen(req)
-    ret = response.read()
-    print ret
-
-################################################################################
-
-def testRedirectWithRssiTxPower(endpoint):
-    values = {
-        'objects': [
-            {
-                'url': endpoint + '/experimental/googl/KYvLwO',
-                'rssi': -41,
-                'txpower': -22,
-                'force': True,
-            },
-            {
-                'url': endpoint + '/experimental/googl/r8iJqW',
-                'rssi': -91,
-                'txpower': -22,
-                'force': True,
-            },
-        ]
-    }
-
-    result = resolveScanForValues(endpoint, values)
-    print json.dumps(result, indent=4)
-
-################################################################################
-
-def testGoLink(endpoint):
-    params = { 'url': 'https://github.com/google/physical-web' }
-    req = urllib2.Request(endpoint + '/go?' + urllib.urlencode(params))
-    response = urllib2.urlopen(req)
-    print response.getcode()
-
-################################################################################
+
+class PwsTest(unittest.TestCase):
+    _HOST = None  # Set in main()
+
+    @property
+    def HOST(self):
+        PwsTest._HOST
+
+    def request(self, params=None, payload=None):
+        PATH = getattr(self, 'PATH')
```

Comment:
Originally I didn't always have it defined, so there was a default value.  I'll change it.